tests/TestCase/ORM/CompositeKeysTest.php

```diff
@@ -254,6 +256,10 @@ public function internalStategiesProvider()
      */
     public function testBelongsToEager($strategy)
     {
+        $assoc = new BelongsTo('Test');
+        if (!$assoc->validStrategy($strategy)) {
```

Comment:
I added those, and sorry for all the commits. I'll squash tomorrow if there are no more changes to make.